scilpy/tractograms/dps_and_dpp_management.py

```diff
@@ -252,26 +254,21 @@ def perform_operation_on_dpp(op_name, sft, dpp_name, endpoints_only=False):
 
     Returns
     -------
-    new_data_per_point: list[np.ndarray]
+    new_data_per_point: ArraySequence
         The values that could now be associated to a new data_per_point key.
     """
     call_op = OPERATIONS[op_name]
     if endpoints_only:
         new_data_per_point = []
         for s in sft.data_per_point[dpp_name]:
-            this_data_per_point = np.nan * np.ones((len(s), 1))
-            this_data_per_point[0] = call_op(s[0])
-            this_data_per_point[-1] = call_op(s[-1])
-            new_data_per_point.append(np.asarray(this_data_per_point)[:, None])
+            this_data_per_point = [call_op(s[0]), call_op(s[-1])]
```

Comment:
This is not good in the sense that what is returned is no longer usable as dpp. The docstring says it should return a new_dpp. 

In test_dps_and_dpp_management, at line 186, you could add a 
`    fake_sft.data_per_point['my_dpp2'] = dpp`
and see that it fails. The previous code was always returning arraysequences of the same length as the streamlines.

Reply:
It is valid as dpp because the return converts everything as a ArraySequence, which is the expected type for dpp (list of array was just used to later initialized an ArraySequence)

Reply:
A more efficient way to store endpoints only operation would be to use dps with a 2D array store per streamline, rather than storing a lot of nan to then only use 2 values. But I will keep that for later.